### PR TITLE
fix: Improve clarity and correctness in showFilterDialog

### LINKED ISSUE #2457: Issue : Filter Selection not working properly
**Summary:**

Filter Selection not working properly

**Steps to reproduce:**

1.Go to accounts
2.select any tranasaction from  account s
3. Go to Transactions
4. select filter

**Expected behaviour:**

It should work properly

**Observed behaviour:**
Not working properly

**Device and Android version:**

**Screenshots:**

https://github.com/openMF/mifos-mobile/assets/136256522/5d774147-381c-4d02-9411-334fd46f5ccc

### PR DESCRIPTION
**Pull request by ghost**
Fixes #2457 

Please Add Screenshots If there are any UI changes.

https://github.com/openMF/mifos-mobile/assets/136256522/53aa2db7-fe6f-4e43-baf5-0e65cd43be0c

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by ghost** [comment]
@PratyushSingh07 review this....Also there was an another issue in this  should i make another issue or solve in this

**Comment by PratyushSingh07** [comment]
> @PratyushSingh07 review this....Also there was an another issue in this should i make another issue or solve in this 

What's the other issue?

**Comment by ghost** [comment]
@PratyushSingh07  The contact us is going to top if there are no transactions it should be at the bottom

**Comment by PratyushSingh07** [comment]
> @PratyushSingh07 The contact us is going to top if there are no transactions it should be at the bottom

Well I personally don't think this is an issue. In my opinion it would have looked odd if it was at the bottom. Why don't you try to change this behaviour and then we can compare which version looks better. Implement it in this PR itself and incase it doesn't look good you can always reset back to the previous commit

**Comment by ghost** [comment]
@PratyushSingh07  Okay!

**Comment by Thanush66** [comment]
@PratyushSingh07 can we an ui if the were no transactions it will display no trasactions and the contact us would be below!

**Comment by PratyushSingh07** [comment]
Hmm ... Your call @Thanush66 I like both of them 😅

**Comment by ghost** [comment]
@PratyushSingh07 ig the second looks cleaner , Its just my opinion..!

**Comment by PratyushSingh07** [comment]
Alright let's go with this

File: app/src/main/res/layout/fragment_saving_account_transactions.xml [hunk 1]
```diff
@@ -8,37 +8,46 @@
...
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:layout_marginStart="@dimen/Mifos.DesignSystem.Spacing.marginWords"
-                android:clickable="true"
-                android:focusable="true"
-                android:text="@string/help_line_number" />
-
-        </LinearLayout>
+            android:gravity="center"
+            android:text="No transactions to display"
```
Comment (PratyushSingh07) [inline app/src/main/res/layout/fragment_saving_account_transactions.xml]: add this to `strings.xml`

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt [hunk 2]
```diff
@@ -310,72 +310,57 @@ class SavingAccountsTransactionFragment : BaseFragment() {
...
+
             if (!isChecked) {
                 isReady = false
                 radioGroupFilter?.clearCheck()
                 selectedRadioButtonId = -1
             } else {
                 if (selectedRadioButtonId == -1) {
-                    val btn = dialogView.findViewById<RadioButton>(R.id.rb_date)
-                    btn.isChecked = true
+                    val btn = dialogView?.findViewById<RadioButton>(R.id.rb_date)
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt]: why havent we used viewbinding here ? Use binding wherever needed

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt [hunk 3]
```diff
@@ -310,72 +310,57 @@ class SavingAccountsTransactionFragment : BaseFragment() {
...
             }
         }
 
-        // restore prev state
-        checkBoxPeriod?.isChecked = isCheckBoxPeriod
-        if (selectedRadioButtonId != -1) {
-            val btn = dialogView?.findViewById<RadioButton>(selectedRadioButtonId)
-            btn?.isChecked = true
-        }
         val checkBoxRecyclerView: RecyclerView? = dialogView?.findViewById(R.id.recycler_view)
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt]: viewbinding here as well

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]
Your commit message is weird. It is a fix and not refactor . Make sure to change your commit message as well

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt (hunk 2 above)
Comment (Thanush66) [reply app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt:328]: what should i name the variable of binding..?

**Comment by ghost** [comment]
@PratyushSingh07 Done with changes.....Any more changes...?

File: app/src/main/res/layout/fragment_saving_account_transactions.xml (hunk 1 above)
Comment (PratyushSingh07) [reply app/src/main/res/layout/fragment_saving_account_transactions.xml]: How many times will I have to tell you not to resolve conversations on your own ?

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt [hunk 4]
```diff
@@ -298,84 +299,70 @@ class SavingAccountsTransactionFragment : BaseFragment() {
     /**
      * Shows a filter dialog
      */
+     private lateinit var filterBinding: LayoutFilterDialogBinding
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt:302]: make this global and change the name of variable to layoutFilterDialogBinding

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt [hunk 5]
```diff
@@ -298,84 +299,70 @@ class SavingAccountsTransactionFragment : BaseFragment() {
...
-        val inflater = activity?.layoutInflater
-        val dialogView = inflater?.inflate(R.layout.layout_filter_dialog, null, false)
-        val checkBoxPeriod: AppCompatCheckBox? = dialogView?.findViewById(R.id.cb_select)
-        val radioGroupFilter = dialogView?.findViewById<RadioGroup>(R.id.rg_date_filter)
-        tvStartDate = dialogView?.findViewById(R.id.tv_start_date)
-        tvEndDate = dialogView?.findViewById(R.id.tv_end_date)
+        filterBinding = LayoutFilterDialogBinding.inflate(layoutInflater)
+        val dialogView = filterBinding.root
+
+        val checkBoxPeriod: AppCompatCheckBox? = filterBinding.cbSelect
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt:307]: do we need to explicitly declare AppCompatCheckBox ?

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/fragment_saving_account_transactions.xml (hunk 1 above)
Comment (ghost) [reply app/src/main/res/layout/fragment_saving_account_transactions.xml]: Oops! My bad didn't know that reviewer should close them...

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt (hunk 4 above)
Comment (ghost) [reply app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt:302]: Okay...needed to change the first letter to Lowercase..?

File: app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt (hunk 5 above)
Comment (ghost) [reply app/src/main/java/org/mifos/mobile/ui/fragments/SavingAccountsTransactionFragment.kt:307]: Nope ig...should i remove it..?

**Pull request closed without merging**